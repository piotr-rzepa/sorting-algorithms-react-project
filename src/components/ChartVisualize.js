/*
 * Komponent bezklasowy będacy wykresem słupkowym z indeksem na osi X oraz wartością na osi Y\
 * Służy do wizualizacji sortowania, którą może zobaczyć użytkownik
 */

/* eslint-disable no-undef */
/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
import React from 'react';
import {
	BarChart,
	Bar,
	XAxis,
	YAxis,
	Legend,
	CartesianGrid,
	ResponsiveContainer
} from 'recharts';

export default function ChartVisualize(props) {
	return (
		<div className="card-chart--big chart-visualize">
			<ResponsiveContainer>
				<BarChart data={props.data} barGap={0} barCategoryGap={0}>
					<CartesianGrid strokeDasharray="3 3" />
					<XAxis dataKey="name" />
					<YAxis />
					<Legend align="right" verticalAlign="top" />
					<Bar dataKey="pv" fill="#8884d8" />
				</BarChart>
			</ResponsiveContainer>
		</div>
	);
}
