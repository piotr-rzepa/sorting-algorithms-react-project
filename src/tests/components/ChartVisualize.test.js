import * as React from 'react';
import { shallow } from 'enzyme';
import ChartVisualize from '../../components/ChartVisualize';

describe('<ChartVisualize />', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<ChartVisualize />);
		expect(wrapper).toMatchSnapshot();
	});
});
