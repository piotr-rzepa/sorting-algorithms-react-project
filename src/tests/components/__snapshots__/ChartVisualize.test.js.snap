// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartVisualize /> renders correctly 1`] = `
<div
  className="card-chart--big chart-visualize"
>
  <ResponsiveContainer
    debounce={0}
    height="100%"
    width="100%"
  >
    <BarChart
      barCategoryGap={0}
      barGap={0}
      layout="horizontal"
      margin={
        Object {
          "bottom": 5,
          "left": 5,
          "right": 5,
          "top": 5,
        }
      }
      reverseStackOrder={false}
      stackOffset="none"
      syncMethod="index"
    >
      <CartesianGrid
        fill="none"
        horizontal={true}
        horizontalFill={Array []}
        horizontalPoints={Array []}
        stroke="#ccc"
        strokeDasharray="3 3"
        vertical={true}
        verticalFill={Array []}
        verticalPoints={Array []}
      />
      <XAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        dataKey="name"
        domain={
          Array [
            0,
            "auto",
          ]
        }
        height={30}
        hide={false}
        mirror={false}
        orientation="bottom"
        padding={
          Object {
            "left": 0,
            "right": 0,
          }
        }
        reversed={false}
        scale="auto"
        tickCount={5}
        type="category"
        width={0}
        xAxisId={0}
      />
      <YAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        domain={
          Array [
            0,
            "auto",
          ]
        }
        height={0}
        hide={false}
        mirror={false}
        orientation="left"
        padding={
          Object {
            "bottom": 0,
            "top": 0,
          }
        }
        reversed={false}
        scale="auto"
        tickCount={5}
        type="number"
        width={60}
        yAxisId={0}
      />
      <Legend
        align="right"
        iconSize={14}
        layout="horizontal"
        verticalAlign="top"
      />
      <Bar
        animationBegin={0}
        animationDuration={400}
        animationEasing="ease"
        data={Array []}
        dataKey="pv"
        fill="#8884d8"
        hide={false}
        isAnimationActive={true}
        layout="vertical"
        legendType="rect"
        minPointSize={0}
        xAxisId={0}
        yAxisId={0}
      />
    </BarChart>
  </ResponsiveContainer>
</div>
`;
